File: src/services/audit/detectors/ad/advanced/signing/smb-v1-enabled.ts
/**
 * SMBv1 Enabled Detector
 * Check if SMBv1 is enabled
 * Requires GPO settings from SYSVOL
 */

import { ADDomain } from '../../../../../../types/ad.types';
import { Finding } from '../../../../../../types/finding.types';
import { GpoSecuritySettings } from '../types';

export function detectSmbV1Enabled(
  gpoSettings: GpoSecuritySettings | null,
  domain: ADDomain | null,
  includeDetails: boolean
): Finding {
  if (gpoSettings && (gpoSettings.smbv1ServerEnabled !== undefined || gpoSettings.smbv1ClientEnabled !== undefined)) {
    const smbv1Enabled = gpoSettings.smbv1ServerEnabled === true || gpoSettings.smbv1ClientEnabled === true;

    return {
      type: 'SMB_V1_ENABLED',
      severity: 'high',
      category: 'advanced',
      title: 'SMBv1 Protocol Enabled',
      description:
        'SMBv1 protocol is enabled. SMBv1 is deprecated and vulnerable to attacks like EternalBlue (WannaCry, NotPetya).',
      count: smbv1Enabled ? 1 : 0,
      affectedEntities: includeDetails && smbv1Enabled && domain ? [domain.dn] : undefined,
      details: smbv1Enabled
        ? {
            recommendation: 'Disable SMBv1 on all systems. Use SMBv2/v3 instead.',
            smbv1Server: gpoSettings.smbv1ServerEnabled,
            smbv1Client: gpoSettings.smbv1ClientEnabled,
          }
        : undefined,
    };
  }

  return {
    type: 'SMB_V1_ENABLED',
    severity: 'high',
    category: 'advanced',
    title: 'SMBv1 Configuration Unknown',
    description: 'Unable to determine SMBv1 configuration. Manual review recommended.',
    count: 0,
    details: {
      note: 'GPO/Registry settings not available. Check SMB1 registry values and Windows features manually.',
    },
  };
}

File: internal/audit/detectors/ad/advanced/signing/smb-v1-enabled.go
package signing

import (
	"context"

	"github.com/etcsec-com/etc-collector/internal/audit"
	"github.com/etcsec-com/etc-collector/pkg/types"
)

// SmbV1EnabledDetector detects if SMBv1 is enabled
type SmbV1EnabledDetector struct {
	audit.BaseDetector
}

// NewSmbV1EnabledDetector creates a new detector
func NewSmbV1EnabledDetector() *SmbV1EnabledDetector {
	return &SmbV1EnabledDetector{
		BaseDetector: audit.NewBaseDetector("SMB_V1_ENABLED", audit.CategoryAdvanced),
	}
}

// Detect executes the detection
func (d *SmbV1EnabledDetector) Detect(ctx context.Context, data *audit.DetectorData) []types.Finding {
	// This requires GPO/registry settings - for now, flag for manual review
	finding := types.Finding{
		Type:        d.ID(),
		Severity:    types.SeverityHigh,
		Category:    string(d.Category()),
		Title:       "SMBv1 Configuration Unknown",
		Description: "Unable to determine SMBv1 configuration. Manual review recommended.",
		Count:       0,
		Details: map[string]interface{}{
			"note":           "GPO/Registry settings not available via LDAP. Check SMB1 registry values and Windows features manually.",
			"recommendation": "Disable SMBv1 on all systems. Use SMBv2/v3 instead.",
			"vulnerabilities": []string{
				"EternalBlue (MS17-010)",
				"WannaCry ransomware",
				"NotPetya malware",
			},
		},
	}

	return []types.Finding{finding}
}

func init() {
	audit.MustRegister(NewSmbV1EnabledDetector())
}
